### PR TITLE
added check for email provider check before sending reset password email

### LINKED ISSUE #1754: Respond with an appropriate error message when the email provider is not configured
While resetting the password, if the email provider is not configured the user should be notified to contact the admin

### PR DESCRIPTION
**Pull request by praffq**
## Proposed Changes

- Added a check to check if the email have valid configrations

### Associated Issue

- Fixes #1754 

image

## Merge Checklist

- [x] Tests added/fixed
- [x] Linting Complete

_*Only PR's with test cases included and passing lint and test pipelines will be reviewed*_

@ohcnetwork/care-backend-maintainers @ohcnetwork/care-backend-admins

### CODE REVIEW

File: care/users/tests/test_auth.py [hunk 1]
```diff
@@ -127,6 +127,34 @@ def test_forgot_password_with_valid_input(self):
...
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(
+            response.json()["detail"][0],
+            "There was a problem resetting your password. Please contact the administrator.",
+        )
+
+    @override_settings(
+        IS_PRODUCTION=True,
+        EMAIL_HOST="smtp.gmail.com",
```
Comment (vigneshhari) [inline care/users/tests/test_auth.py]: use a dummy host not something that could actually work.

File: care/users/tests/test_auth.py [hunk 2]
```diff
@@ -127,6 +127,34 @@ def test_forgot_password_with_valid_input(self):
...
+        EMAIL_HOST_USER="your-email@gmail.com",
+        EMAIL_HOST_PASSWORD="your-app-password",
+    )
+    def test_forgot_password_with_email_configration(self):
+        response = self.client.post(
```
Comment (vigneshhari) [inline care/users/tests/test_auth.py:162]: We should mock the bits that actually send the email and confirm that the email has been sent.

**Review by vigneshhari — CHANGES_REQUESTED** [review]

**Comment by praffq** [comment]
@vigneshhari  made the updates , Please take a look

**Review by sainak — APPROVED** [review]

**Comment by vigneshhari** [comment]
Tests that check if the email was sent were still not added.

**Pull request merged**